src/System.IO.FileSystem/tests/Directory/EnumerableTests.cs

```diff
@@ -32,6 +32,32 @@ public void FileEnumeratorIsThreadSafe()
             }
         }
 
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void EnumerateDirectories_NonBreakingSpace()
+        {
+            var expectedDirectoryNames = new List<string> { GetTestFileName(), "\u00A0", GetTestFileName() };
+
+            DirectoryInfo di = new DirectoryInfo(Path.Combine(TestDirectory, expectedDirectoryNames[0]));
+            di.Create();
+            di.CreateSubdirectory(expectedDirectoryNames[1]);
+            di.CreateSubdirectory(expectedDirectoryNames[1] + Path.DirectorySeparatorChar + expectedDirectoryNames[2]);
+            
+            var actualDirectoryNames = new List<string>();
```

Comment:
This is more complicated than it needs to be. If you really care about the root directory name (which I wouldn't), check it separately. The rest you can and should actually validate the full name, not just the name.

``` C#
DirectoryInfo rootDirectory = Directory.Create(GetTestFilePath());
DirectoryInfo subDirectory1 = rootDirectory.CreateSubdirectory("\u00A0");
DirectoryInfo subDirectory2 = rootDirectroy.CreateSubdirectory(Path.Combine(subDirectory1.Name, "B"));

Assert.Equal(new string[] { subDirectory1.FullName, subDirectory2.FullName }, // Either just use the recursive option in Enumerate or write a recursive function that returns IEnumerable<string> and yields
```